analog_io_modules/seven_sensor_rain_gauge/firmware.lua

```diff
@@ -0,0 +1,191 @@
+      telemetry['Rain Gauge(Hour)']= tostring(data)
+    else
+      enapter.log('Register 40020 reading failed: ' .. modbus.err_to_str(result), 'error')
+      status = 'read_error'
+    end
+  end
+
+  telemetry.alerts = alerts
+  telemetry.status = status
+  enapter.send_telemetry(telemetry)
+end
+
+-- Holds global device connection
+
+
+local device
+
+function connect_device()
+  if device then
+    return device, nil
+  end
+
+  local values, err = config.read_all()
+  if err then
+    enapter.log('cannot read config: ' .. tostring(err), 'error')
+    return nil, 'cannot_read_config'
+  else
+    local address, baudrate, parity, stop_bits, data_bits, serial_port = values[ADDRESS], values[BAUDRATE], values[STOP_BITS], values[PARITY], values[DATA_BITS], values[SERIAL_PORT]
+    if not address or not baudrate or not parity or not stop_bits or not data_bits or not serial_port then
+      return nil, 'not_configured'
+    else
+      -- Declare global variable to reuse connection between function calls
+    --'/dev/ttyS0', {baud_rate=9600,parity="N",stop_bits=1,data_bits=8,read_timeout=1000}
+     device = RainGaugeModbusRtu.new(address,tonumber(baudrate), tostring(parity), tonumber(stop_bits), tonumber(data_bits), tostring(serial_port))
+     device:connect()
+      return device, nil
+    end
+  end
+end
+
+-------------------------------------------
+-- Seven Sensor Solutions Rain Gauge Modbus RTU API
+-------------------------------------------
+
+RainGaugeModbusRtu = {}
+
+--"/dev/ttyS0", {baud_rate=9600,parity="E",stop_bits=1,data_bits=8,read_timeout=1000}
+
+function RainGaugeModbusRtu.new(addr, baudrate, parity, stop_bits, data_bits, serial_port)
+  assert(type(addr) == 'string', 'addr (arg #1) must be string, given: ' .. inspect(addr))
+  assert(type(baudrate) == 'number', 'baudrate (arg #2) must be number, given: ' .. inspect(baudrate))
+  assert(type(parity) == 'string', 'parity (arg #3) must be string, given: ' .. inspect(parity))
+  assert(type(stop_bits) == 'number', 'stop_bits (arg #4) must be number, given: ' .. inspect(stop_bits))
+  assert(type(data_bits) == 'number', 'data_bits (arg #5) must be number, given: ' .. inspect(data_bits))
+  assert(type(serial_port) == 'string', 'serial_port (arg #6) must be string, given: ' .. inspect(serial_port))
+  local self = setmetatable({}, { __index = RainGaugeModbusRtu })
+  self.addr = addr
+  self.baudrate = baudrate
+  self.parity = parity
+  self.stop_bits = stop_bits
+  self.data_bits = data_bits
+  self.serial_port = serial_port
+  return self
+end
+
+function RainGaugeModbusRtu:connect()
+  self.modbus = modbusrtu.new(self.addr)
+end
+
+function RainGaugeModbusRtu:read_inputs(address, number)
+  assert(type(address) == 'number', 'address (arg #1) must be number, given: ' .. inspect(address))
+  assert(type(number) == 'number', 'number (arg #2) must be number, given: ' .. inspect(number))
+
+  local registers, err = self.modbus:read_inputs(self.address, number, 1000)
```

Comment:
Signature must be the same as in ["classic" Modbus RTU](https://developers.enapter.com/docs/reference/ucm/modbus#modbusread_inputs)